ckanext/restricted/action.py

```diff
@@ -90,7 +89,9 @@ def resource_view_list(resource_view_list, context, data_dict):
 
 
 @toolkit.side_effect_free
-def restricted_package_show(context, data_dict, package_metadata=None):
+def restricted_package_show(context, data_dict, package_metadata=None,
+                            user_is_package_collaborator_cache={}, user_can_update_package_cache={},
```

Comment:
Thanks, good catch, I still have a lot of python quirks ahead of me ;) !

Funny thing - when I used your approach my code... slowed down (and the test caught it! :grinning: ). One should test:
`if a is None:
   a = {}`

since 'not' evaluates to true even when given object exists but is empty.